### PR TITLE
feat: disallow non-ascii characters in desc field

### PR DESCRIPTION
**Pull request by JasonChong96**
## Problem

When users input descriptions, they're allowed to enter non-ascii characters such as chinese characters. Characters from other languages are not effectively indexed for search. Furthermore, GoGovSg is monolingual at the moment and visitors are unlikely to not know English.

## Solution

Disallow non-ascii characters on both the input field and backend API validation.

## Before & After Screenshots

**AFTER**:

image

## Tests

There should be an error message and the save button greyed out when a non-english character (e.g chinese) is inputted in the description field.

### CODE REVIEW

**Comment by LoneRifle** [comment]
The error message may be confusing to non-technical users (ie, "what is ASCII?"). Could you rephrase this?

**Comment by JasonChong96** [comment]
> The error message may be confusing to non-technical users (ie, "what is ASCII?"). Could you rephrase this?

Would saying that only english characters are allowed be less confusing?

**Comment by LoneRifle** [comment]
We might need to have the designers involved. @pearlyong - thoughts?

**Comment by pearlyong** [comment]
Suggested:
Description cannot contain special characters

**Comment by JasonChong96** [comment]
Changed to "Description should only contain alphanumeric characters and symbols" as per discussion.

File: src/shared/util/validation.ts [hunk 1]
```diff
@@ -58,3 +58,7 @@ export function isCircularRedirects(url: string, hostname?: string): boolean {
   if (!hostname) return false
   return parse(url).hostname === hostname
 }
+
+export function isPrintableAscii(string: string): boolean {
+  return /^[\x20-\x7F]*$/.test(string)
+}
```
Comment (liangyuanruo) [inline src/shared/util/validation.ts:64]: can we document this function to say that this is merely the hexadecimal representation of the ascii charset
